CorsixTH/Lua/entities/humanoids/vip.lua

```diff
@@ -202,268 +290,240 @@ function Vip:announce()
+      num_cons = 0
+    end
+    local num_junior = self.hospital:hasStaffOfCategory("Junior")
+    if not num_junior then
+      num_junior = 0
+    end
 
-  if self.num_visited_rooms ~= 0 then
-    self.vip_rating = self.vip_rating + self.room_eval / self.num_visited_rooms
+    -- check consultant and junior proportions
+    if num_cons / num_docs > 0.5 then
+      self.vip_rating = self.vip_rating - 1
+    elseif num_junior / num_docs > 0.5 then
+      self.vip_rating = self.vip_rating + 1
+    end
   end
 
-  -- check average patient happiness
-  local avg_happiness = self.hospital:getAveragePatientAttribute("happiness", nil)
-  if avg_happiness then
-    local patients_happy_ratio_rangemap = {
-      {upper = 0.20, value = -10},
-      {upper = 0.40, value = -5},
-      {upper = 0.60, value = 0},
-      {upper = 0.80, value = 5},
-                    {value = 10}
+--[[--Group factor 5: Rooms--]]
+  -- Low room numbers incur a penalty
+  if count_rooms < 1 then
+    self.vip_rating = self.vip_rating + 4
+  else
+    if count_rooms < 3 then
+      self.vip_rating = self.vip_rating + 1
+    end
+    -- Room decor average
+    local avg_room_eval = self.room_eval / self.num_visited_rooms
+    local room_eval_rangemap = {
+      {upper = 2, value = 3},
+      {upper = 3, value = 1},
+      {upper = 4, value = 0},
+      {value = -1}
     }
-    self.vip_rating = self.vip_rating + rangeMapLookup(avg_happiness, patients_happy_ratio_rangemap)
+    if self.num_visited_rooms ~= 0 then
+      self.vip_rating = self.vip_rating + rangeMapLookup(avg_room_eval, room_eval_rangemap)
+    end
   end
 
-  -- check average staff happiness
-  avg_happiness = self.hospital:getAverageStaffAttribute("happiness", nil)
-  if avg_happiness then
-    local staff_happy_ratio_rangemap = {
-      {upper = 0.20, value = -10},
-      {upper = 0.40, value = -5},
-      {upper = 0.60, value = 0},
-      {upper = 0.80, value = 5},
-                    {value = 10}
-    }
-    self.vip_rating = self.vip_rating + rangeMapLookup(avg_happiness, staff_happy_ratio_rangemap)
+--[[--Finalise score--]]
+  --documented rewards.
+  local rewards = {
+    [1] = 4000,
+    [2] = 2000,
+    [3] = 1500,
+    [4] = 1200,
+    [5] = 800,
+    [6] = 400,
+    [7] = 200,
+    [8] = 0,
+  }
+--documented reps
+  local rep_change = {
+    [1] = 50,
+    [2] = 45,
+    [3] = 40,
+    [4] = 35,
+    [5] = 30,
+    [6] = 25,
+    [7] = 20,
+    [8] = 15,
+    [9] = 10,
+    [10] = 5,
+    [11] = -5,
+    [12] = -10,
+    [13] = -15,
+    [14] = -20,
+    [15] = -25,
+  }
+  -- apply caps
+  if self.vip_rating >= 8 then
+    self.cash_reward = rewards[8]
+  elseif self.vip_rating <= 1 then
+    self.cash_reward = rewards[1]
+  else
+    self.cash_reward = rewards[self.vip_rating]
   end
-
-  -- set the cash reward value
-  if tonumber(self.world.map.level_number) then
-    self.cash_reward = math.round(self.world.map.level_number * self.vip_rating) * 10
+  if self.vip_rating >= 15 then
```

Comment:
If you wanted to save code in this block:
```lua
-- bound vip rating
self.vip_rating = self.vip_rating > 15 and 15 or self.vip_rating
self.vip_rating = self.vip_rating < 1 and 1 or self.vip_rating
self.cash_reward = rewards[self.vip_rating] or 0
self.rep_reward = rep_change[self.vip_rating]
self.vip_message = self.vip_rating -- not sure why you want a different value here
```

Reply:
Added for next revision